src/main.rs

```diff
@@ -269,14 +275,17 @@ impl RpcService {
                 Box::new(self.generate_work(root, difficulty).then(move |res| match res {
                     Ok(work) => {
                         let end = PreciseTime::now();
-                        println!("work_generate completed in {}ms for difficulty {:#x}",
+                        writeln!(io::stdout(), "work_generate completed in {}ms for difficulty {:#x}",
                             start.to(end).num_milliseconds(),
-                            difficulty);
-                        let work: Vec<u8> = work.iter().rev().cloned().collect();
+                            difficulty)
+                            .unwrap_or(());
```

Comment:
Idiomatic Rust is usually `let _ = writeln!(...);` to throw away errors. I'd use that instead.

Reply:
Thank you! Interestingly that workaround is not reported in https://github.com/rust-lang/rust/issues/24821

Reply:
Correct about the vector, reverted the change/

Reply:
For the vector I think it should actually use https://doc.rust-lang.org/std/vec/struct.Vec.html#method.reverse instead, which is a lot clearer about what it does.